Add 2.10 upgrade instructions. (#872)
digital-asset/docs.daml.com: docs/2.10.0/docs/canton/usermanual/upgrading.rst

If Canton starts and shows the command prompt of the console, then the configuration was
parsed successfully.

.. note::

>     The command line option ``--manual-start`` prevents the node from starting up automatically,
>     as we first need to migrate the database.

Reviewer: ```suggestion
    The command line option ``--manual-start`` prevents the node from starting automatically,
    allowing you to migrate the database.
```

.. _migrating_the_database:

Migrating the Database
~~~~~~~~~~~~~~~~~~~~~~

> Canton does not perform a database migration automatically. Migrations need to be forced. Therefore, it is
> recommended to start Canton using the command line option ``--manual-start`` (see :ref:`above <test-your-config>`)
> when upgrading to a new Canton binary version that requires a database migration.
> 
> If you start a node that requires a database migration, you will observe the following Flyway error:

Reviewer: ```suggestion
Canton does not perform a database migration automatically. Migrations must be forced. We recommend that you start Canton using the command line option ``--manual-start`` (see :ref:`above <test-your-config>`)
when upgrading to a new Canton binary version that requires a database migration.

If you start a node that requires a database migration, you will observe the following Flyway error:
```

Version Specific Notes
~~~~~~~~~~~~~~~~~~~~~~

.. _upgrade_to_2.10:

Upgrade to Release 2.10
^^^^^^^^^^^^^^^^^^^^^^^

Protocol Versions
"""""""""""""""""

> 2.10 is a Long Term Support (LTS) release and introduces a new Canton protocol version 7, which is the

Reviewer 2: I am trying to clarify if we want to call 2.10 an LTS already.

Reviewer 3: I think this is a good point and we don't want to do it right now.

Reviewer 4: yes we don't call it LTS yet, we will do that later once we have confirmed it is stable enough.

protocol version 5, but certain new features (e.g., Smart Contract Upgrading) are only
available in protocol version 7.

> `Please note that protocol version 6 has been marked as deleted and should not be used`.
> Protocol version 7 has been introduced as its stable replacement (see :ref:`here <protocol_version>`
> for more information about protocol versions).

Reviewer: ```suggestion
`Note that protocol version 6 has been marked as deleted and should not be used`.
Protocol version 7 has been introduced as its stable replacement (see :ref:`here <protocol_version>`
for more information about protocol versions).
```

Protocol version should be set explicitly
"""""""""""""""""""""""""""""""""""""""""

> Recall that since the 2.9 release, you must set the protocol version explicitly.

Reviewer: ```suggestion
Since the 2.9 release, you must set the protocol version explicitly.
```

To specify the protocol version for your sync domain:

.. code:: text

    myDomain {
        init.domain-parameters.protocol-version = 7
    }

For a domain manager:

.. code:: text

    domainManager {
        init.domain-parameters.protocol-version = 7
    }

> Please ensure all environments use the same protocol version: you should not use one protocol version in
> a test environment and another one in production.

Reviewer: ```suggestion
Make sure all environments use the same protocol version: you should not use one protocol version in
a test environment and another one in production.
```

> If a protocol version is not provided, then an error message like this will be generated:

Reviewer: ```suggestion
If a protocol version is not provided, you receive an error message:
```

.. code:: text

    ERROR c.d.c.CantonEnterpriseApp$ - CONFIG_VALIDATION_ERROR(8,0): Failed to validate the configuration due to:
    Protocol version is not defined for domain `mydomain`. Define protocol version at key `init.domain-parameters.protocol-version`...

Enabling Smart Contract Upgrades
""""""""""""""""""""""""""""""""

> As mentioned above, the protocol version must be set to 7 in order to utilize Smart
> Contract Upgrades. For existing systems, this requires a domain migration to protocol version 7
> when rolling out the 2.10 Canton binary.

Reviewer: ```suggestion
As mentioned above, the protocol version must be set to 7 to use Smart
Contract Upgrades. For existing systems, this requires a domain migration to protocol version 7
when rolling out the 2.10 Canton binary.
```

> 

Reviewer 2: This contradicts the release notes

Author: Hopefully should be good now (removed the Beta part)

Reviewer 2: Thank you
